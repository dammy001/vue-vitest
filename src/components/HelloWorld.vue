<script setup lang="ts">
import { ref } from 'vue';

defineProps<{ msg: string }>();

const count = ref<number>(0);
</script>

<template>
  <h1 class="text-3xl font-bold text-center">{{ msg }}</h1>

  <button
    type="button"
    @click="count++"
    class="bg-blue-500 py-2 px-8 hover:ring-blue-600"
  >
    count is: {{ count }}
  </button>
</template>

<style scoped>
a {
  color: #42b983;
}

label {
  margin: 0 0.5em;
  font-weight: bold;
}

code {
  background-color: #eee;
  padding: 2px 4px;
  border-radius: 4px;
  color: #304455;
}
</style>
