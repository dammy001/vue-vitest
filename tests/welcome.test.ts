import { mount } from '@vue/test-utils';
import HelloWorld from '../src/components/HelloWorld.vue';

test('HelloWorld.vue', async () => {
 expect(HelloWorld).toBeTruthy();

 const wrapper = mount(HelloWorld, {
  props: {
   msg: 'Hello World',
  },
 });

 expect(wrapper.text()).toContain('Hello World');
 expect(wrapper.text()).toContain(0);
 expect(wrapper.html()).toMatchSnapshot();

 await wrapper.get('button').trigger('click');
 expect(wrapper.text()).toContain(1);
});
