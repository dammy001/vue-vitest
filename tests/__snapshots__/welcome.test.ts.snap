// Vitest Snapshot v1

exports[`HelloWorld.vue 1`] = `
"<h1 data-v-469af010=\\"\\">Hello World</h1>
<button type=\\"button\\" data-v-469af010=\\"\\">count is: 0</button>"
`;
